### DISCUSSION TITLE
Need example on compressing image (without resizing)

### DISCUSSION THREAD

**Discussion by Sparkenstein** [image-rs/image#2218] [Q&A]
I was looking for something like https://squoosh.app/, but written in rust, something that only compresses images without resizing them. is there any simple example I can get started with? one example in examples dir resizes the image

**Comment by Sparkenstein** [comment]
I found `JpegEncoder::new_with_quality`, is that it? but this doesn't work for png then. 
libvips has a neat `save_buffer` api which accepts quality arg for all methods for example:

https://www.libvips.org/API/8.7/VipsForeignSave.html#vips-pngsave-buffer

**Reply by fintelia** [reply]
There's also `PngEncoder::new_with_quality`. But squoosh [internally uses opipng and mozjpeg (among other codecs)](https://github.com/GoogleChromeLabs/squoosh/tree/dev/codecs) which will outperform the encoders in this library. You might want look into using those instead

**Reply by Sparkenstein** [reply]
what about webp? webp crate has an option to use quality arg. but imagers crate doesn't support quality for webp. Ideally I would like the experience to be same for all formats. I don't see any other option besides using multiple crates and handling conversion manually

**Reply by fintelia** [reply]
We don't use the `webp` crate anymore. We now use `image-webp` which hasn't yet added support for a controllable quality setting (PRs welcome!)

**Reply by antonio-castellon** [reply]
what about TIFF compression modes? 
We would like to compare different compression outputs (size and time consuming as well).

An example about how to select the compression type and/or quality will be great.

Many thanks in advance for your answers,
Antonio

**Reply by fintelia** [reply]
At the moment, I think our TIFF encoder doesn't support compression. You can see the capabilities of different encoders by looking at [docs.rs page for the codecs module](https://docs.rs/image/latest/image/codecs/index.html) (scroll down to the bottom of that page and click the format you are interested in)

**Reply by spoutn1k** [reply]
I recently opened a PR to add LZW compression support for TIFF images, getting pretty good results on my personal images, with results comparable to ImageMagick.

**Comment by houseme** [comment]
Image-webp to create a picture will be much larger than jpg, how to adjust the quality of the picture? You can adjust the quality value like [webp](https://crates.io/crates/webp) crate.
